### DISCUSSION TITLE
File truncate and append over MQTT

### DISCUSSION THREAD

**Discussion by PhantomSage** [HASwitchPlate/openHASP#482] [Ideas]
**Is your feature request related to a problem? Please describe.**
Sometimes it is impossible to address the plate via direct IP, but MQTT works due to network topology.
I now have plates on 2 locations on networks connected to a common mqtt server, 
Thus, it would be nice to rewrite pages.jsonl and the other files via MQTT commands.

**Describe the solution you'd like**

`command/filetrunc pages.jsonl`
`command/fileappend pages.jsonl ["row 1","row 2"]`
`command/fileappend pages.jsonl ["row 3","row 4"]`

This would allow for modifying multiple files assuming they are text based.
Would cover the case with online/offline/pages atleast.

**Describe alternatives you've considered**
An alternative is to have a "savepages" command, but that's not equally generic.
Or sending the pages directly to ram via MQTT, but not efficient nor persistent.

**Additional context**
Not really high prior, just ran into a bottleneck and thought I'd share the problem and a potential solution.

**Comment by fvanroie** [comment]
There are currently no plans to implement file management commands over MQTT. 😞 

The File Manager is WebUI only, so you need to figure out a way to access the plate using HTTP (like a VPN).
Alternatively, you can implement custom MQTT commands yourself by adding [Custom Code in the firmware](https://www.openhasp.com/0.7.0/firmware/compiling/custom/).
